File: 75-sort-colors/75-sort-colors.scala
import scala.collection.mutable._

object Solution {
  def sortColors(nums: Array[Int]): Unit = {
    val cn = Array(0, 0, 0)
    nums.foreach { num => cn(num) += 1 }

    var j = 0
    for (i <- 0 to nums.length - 1) {
      while (cn(j) == 0) j += 1
      cn(j) -= 1
      nums(i) = j
    }
  }
}

File: 75-sort-colors/75-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int l = 0;
        int r = nums.length - 1;
        int i = 0;
        while(i <= r) {
            if (nums[i] == 0) swap(nums, i++, l++);
            else if (nums[i] == 2) swap(nums, i, r--);
            else i++;
        }
    }
    
    void swap(int[] nums, int a, int b) {
        int tmp = nums[a];
        nums[a] = nums[b];
        nums[b] = tmp;
    }
}